integration-tests/utils/integration-helpers.js

```diff
@@ -217,6 +218,22 @@ const withWorkspace = (test) => async (options) => {
   }
 };
 
+const withProtectedWorkspace = (test) => async (options) => {
+  console.log('withWorkspace ...');
+  const { workspaceName } = await makeGcpWorkspace({ ...options, isProtected: true });
+
+  try {
+    await test({ ...options, workspaceName });
+  } finally {
+    console.log('withWorkspace cleanup ...');
+    const didDelete = await withSignedInPage(deleteWorkspaceInUi)({ ...options, workspaceName });
+    if (!didDelete) {
+      // Pass test on a failed cleanup - expect leaked resources to be cleaned up by the test `delete-orphaned-workspaces`
+      console.error(`Unable to delete workspace ${workspaceName} via the UI. The resource will be leaked!`);
+    }
+  }
+};
+
```

Comment:
instead of `withProtectedWorkspace` containing an implementation, could this just set `isProtected: true` in the options and call `withWorkspace`?

Reply:
Why, yes it can. Done in https://github.com/DataBiosphere/terra-ui/pull/4769/commits/600ca73a12fd5b9ec0c8a2fa4f36f8a198960c0f.